### PR TITLE
improve tutorial 4 recommend deploy contract first 

### PR DESCRIPTION
**Pull request by qwadratic**
[Tutorial 4](https://docs.minaprotocol.com/zkapps/tutorials/zkapp-ui-with-react) uses the [hardcoded zkApp address](https://github.com/o1-labs/docs2/blob/e2248ea9b804a64f1951727b02b908fdb71ad4fb/examples/zkapps/04-zkapp-browser-ui/ui/src/pages/index.page.tsx#L79), but due to Berkeley reloads, it's better to add a note that a user should deploy contract first. Also zkapp address is better to be defined as const on top of the `index.tsx` file

### CODE REVIEW

File: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx [hunk 1]
```diff
@@ -248,40 +248,47 @@ First, review the `import` and `export` statements:
...
+const ZKAPP_ADDRESS = 'B62qo2Be4Udo5EG1ux9yMJVkXe9Gz945cocN7Bn4W9DSYyeHZr1C3Ea';
+
+export default function Home() {
+  return <div/>
+}
 ```
 
+:::info
+
+You will likely need to deploy the zkApp to Berkeley yourself, as this network is sometimes being reloaded. When deployed, replace the `ZKAPP_ADDRESS` with the address of your deployment.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:268]:
```suggestion
For best results, deploy the zkApp to Berkeley yourself, as this test network is sometimes being reloaded. When deployed, replace `ZKAPP_ADDRESS` with the address of your deployment.
```

File: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx [hunk 2]
```diff
@@ -248,40 +248,47 @@ First, review the `import` and `export` statements:
...
- 15     accountExists: false,
- 16     currentNum: null as null | Field,
- 17     publicKey: null as null | PublicKey,
- 18     zkappPublicKey: null as null | PublicKey,
- 19     creatingTransaction: false,
- 20   });
- 21 
- 22   const [displayText, setDisplayText] = useState('');
- 23   const [transactionlink, setTransactionLink] = useState('');
+export default function Home() {
```
Comment (barriebyron) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:278]: thanks for these updates @qwadratic 
I defer to  @ymekuria for the code update review and to verify that the tutorial runner does not have line number dependencies

Comment (barriebyron) [reply docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:278]: @ymekuria  can you also evaluate https://github.com/o1-labs/docs2/issues/694

File: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx [hunk 3]
```diff
@@ -248,40 +248,47 @@ First, review the `import` and `export` statements:
...
+const ZKAPP_ADDRESS = 'B62qo2Be4Udo5EG1ux9yMJVkXe9Gz945cocN7Bn4W9DSYyeHZr1C3Ea';
+
+export default function Home() {
+  return <div/>
+}
 ```
 
+:::info
+
+For best results, deploy the zkApp to Berkeley yourself, as this test network is sometimes being reloaded. When deployed, replace `ZKAPP_ADDRESS` with the address of your deployment.
```
Comment (ymekuria) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:268]:
```suggestion
The smart contract that the UI interacts with in this tutorial has been deployed to the Berkeley network and the public key is stored as the `ZKAPP_ADDRESS` in the UI project. If you experience problems with the deployed contract, you can deploy the `Add` contract included in the `contracts` folder yourself to the Berkeley network. When deployed, replace `ZKAPP_ADDRESS` with the  public key of your deployment.
```

**Review by ymekuria — CHANGES_REQUESTED** [review]
I left a suggestion. I think it is a good idea to suggest deploying to Berkeley but suggested a change to how we present that option.

**Review by ymekuria — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**